### DISCUSSION TITLE
TOSCA 2.0 support

### DISCUSSION THREAD

**Discussion by cdanger** [tliron/go-puccini#143] [Q&A] [answered]
Regarding TOSCA 2.0 support, the README on the puccini only says _TOSCA 2.0 (in progress)_. Could you give more details about which parts of the [TOSCA 2.0 spec (latest draft)](https://github.com/oasis-tcs/tosca-specs/blob/working/tosca_2_0/TOSCA-v2.0.md) are not yet supported?
Do you keep track of the progress somewhere somehow?
Thanks in advance for any detail on the matter.

**Comment by tliron** [comment, marked as answer]
I have not been keeping progress in an organized way, but the spec is also not fully settled. The big things that have been implemented are the new function style, various keyword renaming, and support for tarball CSARs. Other aspects have still not been entirely finalized in OASIS.

A csd of TOSCA 2.0 is going to be released at the end of the month, and that would be quite close to a final version.

Puccini is going to follow 2.0 quite quickly. I imagine that by the time 2.0 is finalized, Puccini will be ready for 2.0.

**Comment by cdanger** [comment]
Thanks for the heads up, that's a start, looking forward to the next release of TOSCA 2.0 spec :-) .
